Assets/Scripts/Models/Buildable/Components/TemperatureConnection.cs

```diff
@@ -0,0 +1,149 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System;
+using System.Collections.Generic;
+using System.Xml.Serialization;
+using MoonSharp.Interpreter;
+using Newtonsoft.Json;
+using ProjectPorcupine.Localization;
+using ProjectPorcupine.Rooms;
+
+namespace ProjectPorcupine.Buildable.Components
+{
+    /// <summary>
+    /// Outputs heat, but can also be set to require a certain heat to function.
+    /// </summary>
+    [Serializable]
+    [JsonObject(MemberSerialization.OptIn)]
+    [XmlRoot("Component")]
+    [BuildableComponentName("TemperatureConnection")]
+    [MoonSharpUserData]
+    public class TemperatureComponent : BuildableComponent
+    {
+        public TemperatureComponent()
+        {
+        }
+
+        private TemperatureComponent(TemperatureComponent other) : base(other)
+        {
+            Requires = other.Requires;
+        }
+
+        [XmlElement("Requires")]
+        [JsonProperty("Requires")]
+        public TemperatureInfo Requires { get; set; }
+
+        [XmlElement("ParameterDefinitions")]
+        [JsonProperty("ParameterDefinitions")]
+        public TemperatureConnectionParameterDefinitions ParamsDefinitions { get; set; }
+
+        public override bool RequiresSlowUpdate
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        [XmlIgnore]
+        public bool Outputs
+        {
+            get
+            {
+                return FurnitureParams[ParamsDefinitions.OutputRate.ParameterName].ToFloat() != -1;
+            }
+        }
+
+        [XmlIgnore]
+        public float OutputRate
+        {
+            get
+            {
+                return FurnitureParams[ParamsDefinitions.OutputRate.ParameterName].ToFloat();
+            }
+
+            set
+            {
+                FurnitureParams[ParamsDefinitions.OutputRate.ParameterName].SetValue(value);
+            }
+        }
```

Comment:
Does OutputRate needs to be dynamic? Will it be set to different value? If not, you can have just float property OutputRare in here - then in .xml it will be defined inside Component tag.
Or if you want to keep it as furniture param, you can still define if as property, and then during initialization set it though this OutputRate setter in initialization.
This is just to have input related to component have defined inside that component definition..